datatypes/src/operations/reproject.rs

```diff
@@ -214,14 +226,71 @@ where
             .chain(left_line)
             .collect();
 
+        let proj_outline_coordinates = projector.project_coordinates(&outline_coordinates)?;
+
+        Ok(MultiPoint::new_unchecked(proj_outline_coordinates).spatial_bounds())
+    }
+}
+
+pub trait ReprojectClipped<P: CoordinateProjection> {
+    type Out;
+    /// Reproject and clip with respect to the area of use of the projection
+    fn reproject_clipped(&self, projector: &P) -> Result<Self::Out>;
+}
+
+impl<P> ReprojectClipped<P> for BoundingBox2D
+where
+    P: CoordinateProjection,
+{
+    type Out = BoundingBox2D;
+    fn reproject_clipped(&self, projector: &P) -> Result<BoundingBox2D> {
+        const POINTS_PER_LINE: i32 = 7;
+
+        // clip bbox to the area of use of the target projection
+        let area_of_use_projector = CoordinateProjector::from_known_srs(
+            SpatialReference::epsg_4326(),
+            projector.source_srs(),
+        )?;
+        let source_area_of_use = projector.source_srs().area_of_use()?;
+        let target_area_of_use = projector.target_srs().area_of_use()?;
+        let area_of_use = source_area_of_use
+            .intersection(&target_area_of_use)
+            .ok_or(Error::BboxesDoNotIntersect)?;
+        let area_of_use = area_of_use.reproject(&area_of_use_projector)?;
+        let clipped_bbox = self
+            .intersection(&area_of_use)
+            .ok_or(Error::BboxesDoNotIntersect)?;
+
+        // project points on the bbox
```

Comment:
you could just call project on the clipped bbox. It would also be nice if you could move the empty check to the non clipped one.

Reply:
we discussed that there are different error handlings in the two methods. The empty check is not needed in the non-clipped one